### PR TITLE
more cpp lints

### PR DESCRIPTION
**Pull request by guolinke**
refer to #1990

### CODE REVIEW

File: include/LightGBM/utils/common.h [hunk 1]
```diff
@@ -313,6 +313,7 @@ inline static unsigned CountDecimalDigit32(uint32_t n) {
     1000000000
   };
 #ifdef _MSC_VER
+  // NOLINT
```
Comment (guolinke) [inline include/LightGBM/utils/common.h:316]: `_BitScanReverse` only accepts `unsigned long`.

**Comment by StrikerRUS** [comment]
I guess `NOLINT` should be on the same line.
> False positives can be ignored by putting // NOLINT at the end of the line or // NOLINTNEXTLINE in the previous line.
  https://google.github.io/styleguide/cppguide.html#cpplint

File: include/LightGBM/utils/common.h (hunk 1 above)
Comment (guolinke) [inline include/LightGBM/utils/common.h:316]:
```suggestion
  // NOLINTNEXTLINE
```

File: src/network/socket_wrapper.hpp [hunk 2]
```diff
@@ -218,6 +218,7 @@ class TcpSocket {
         continue;
       }
       if (ifa->ifa_addr->sa_family == AF_INET) {
+        // NOLINT
```
Comment (guolinke) [inline src/network/socket_wrapper.hpp:221]:
```suggestion
        // NOLINTNEXTLINE
```

**Comment by StrikerRUS** [comment]
@jameslamb 
We have a new `NOTE` here on macOS
```
* checking examples ... NOTE
Examples with CPU or elapsed time > 5s
     user system elapsed
dim 5.426  0.264    5.79
```

Can you please take a look?

**Comment by jameslamb** [comment]
> @jameslamb
> We have a new `NOTE` here on macOS
> 
> ```
> * checking examples ... NOTE
> Examples with CPU or elapsed time > 5s
>      user system elapsed
> dim 5.426  0.264    5.79
> ```
> 
> Can you please take a look?

😬 wow looks like we have been just barely getting under some of these checks.

I'll take a look. Opened #2988 , should be quick to fix

**Comment by jameslamb** [comment]
> > @jameslamb
> > We have a new `NOTE` here on macOS
> > ```
> > * checking examples ... NOTE
> > Examples with CPU or elapsed time > 5s
> >      user system elapsed
> > dim 5.426  0.264    5.79
> > ```
> > 
> > 
> > Can you please take a look?
> 
> 😬 wow looks like we have been just barely getting under some of these checks.
> 
> I'll take a look. Opened #2988 , should be quick to fix

created #2989 to fix it

**Pull request merged**